File: core/feriado.js
import { addDays, getDate, getMonth } from "date-fns";

export function feriadoComDataFixa(dia, mes, descricao) {
  let feriado = {};

  feriado.dia = dia;
  feriado.mes = mes;
  feriado.descricao = descricao;

  return feriado;
}

export function feriadoComIntervaloFixo(domingoDePascoa, intervalo, descricao) {
  let diaAdicionado = getDate(addDays(domingoDePascoa, intervalo));
  let mesAdicionado = getMonth(addDays(domingoDePascoa, intervalo));

  return {
    diaAdicionado,
    mesAdicionado,
    descricao,
  };
}

// export default function feriado(dia, mes, descricao) {
//   this.dia = dia;
//   this.mes = mes;
//   this.descricao = descricao;

//   this.obterFeriadoComIntervaloFixo = function(domingoDePascoa, intervalo) {
//     let diaAdicionado = getDate(addDays(domingoDePascoa, intervalo));
//     let mesAdicionado = getMonth(addDays(domingoDePascoa, intervalo));

//     let intervaloFixo = new Feriado(
//       diaAdicionado,
//       mesAdicionado,
//       this.descricao
//     );
//     return intervaloFixo;
//   };
// }

File: base/Feriado.cs
using System;

namespace Calendario
{
    public class Feriado
    {
        public int Dia { get; set; }
        public int Mes { get; set; }
        public string Descricao { get; set; }

        public Feriado(int dia, int mes, string descricao)
        {
            Dia = dia;
            Mes = mes;
            Descricao = descricao;
        }
        public static Feriado ObterFeriadoComIntervaloFixo(DateTime DomingoDePascoa, int intervalo, string descricao)
        {
            return new Feriado(DomingoDePascoa.AddDays(intervalo).Day, DomingoDePascoa.AddDays(intervalo).Month, descricao);
        }
    }
}
